third_party/txt/src/txt/font_collection.cc

```diff
@@ -112,57 +123,77 @@ void FontCollection::DisableFontFallback() {
 
 std::shared_ptr<minikin::FontCollection>
 FontCollection::GetMinikinFontCollectionForFamily(
-    const std::string& font_family,
+    const std::vector<std::string>& font_families,
     const std::string& locale) {
   // Look inside the font collections cache first.
-  FamilyKey family_key(font_family, locale);
+  FamilyKey family_key(font_families, locale);
   auto cached = font_collections_cache_.find(family_key);
   if (cached != font_collections_cache_.end()) {
     return cached->second;
   }
 
-  for (sk_sp<SkFontMgr>& manager : GetFontManagerOrder()) {
-    std::shared_ptr<minikin::FontFamily> minikin_family =
-        CreateMinikinFontFamily(manager, font_family);
-    if (!minikin_family)
-      continue;
-
-    // Create a vector of font families for the Minikin font collection.
-    std::vector<std::shared_ptr<minikin::FontFamily>> minikin_families = {
-        minikin_family,
-    };
-    if (enable_font_fallback_) {
-      for (std::string fallback_family : fallback_fonts_for_locale_[locale]) {
-        auto it = fallback_fonts_.find(fallback_family);
-        if (it != fallback_fonts_.end()) {
-          minikin_families.push_back(it->second);
+  const auto default_font_family = GetDefaultFontFamily();
+  size_t font_families_count = font_families.size();
+
+  // Attempt to match each font family in order.
+  for (size_t fallback_index = 0; fallback_index < font_families_count + 1;
+       fallback_index++) {
+    std::string font_family;
+    // When we run out of fonts to attempt to find, try to match the default
+    // font.
+    if (fallback_index == font_families_count) {
+      font_family = default_font_family;
```

Comment:
Previously the engine would try the style's font family in all font managers and then use the default font family if the style's font could not be matched anywhere.

Can we keep that prioritization?  For each family in the fallback list, try it in each font manager.  Only resort to the default font family if none of those combinations work.

Reply:
This change should preserve that behavior. The `fallback_index` increases to font_families.size() + 1, where the extra iteration is caught here and set to the default font at the end when every other option has been exhausted.